### PR TITLE
Add first version of a User Guide

### PR DESCRIPTION
**Pull request by Silvan-WMDE**
For now, this includes three sections about authentication and authorization. To be expanded.

Bug: [T286340](https://phabricator.wikimedia.org/T286340)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Mismatch Finder User Guide
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Logging in <a id="login"></a>](#logging-in-a-idlogina)
+- [Obtaining an API token <a id="apiToken"></a>](#obtaining-an-api-token-a-idapitokena)
+- [Accessing the API <a id="apiAccess"></a>](#accessing-the-api-a-idapiaccessa)
```
Comment (itamargiv) [inline docs/UserGuide.md:8]: I'm pretty sure these `<a>` tags aren't supposed to be here. We also need to make sure that the `#` link is correct. See suggestion.

```suggestion
- [Logging in](#login)
- [Obtaining an API token](#apiToken)
- [Accessing the API](#apiAccess)
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Logging in <a id="login"></a>
+
+You can log in to the Mismatch Finder website using your MediaWiki account on `www.wikidata.org`. Simply click the Login button on the Welcome page and get redirected to Wikidata, to allow Mismatch Finder access to your account as a "Connected Application". If you are not logged in already, Wikidata will ask you for the username and passwort of your MediaWiki account.
```
Comment (itamargiv) [inline docs/UserGuide.md:14]: Nice and concise! Very well put, thank you!

File: docs/UserGuide.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## Obtaining an API token <a id="apiToken"></a>
+
+Once you have logged in and have started to use the Mismatch Finder web interface, you may want to get access to the REST API, as well. You will need a personalised token for that, which you must provide in the `Authorization:` header of your API requests. Open the "API Token" tab to create a token. Mismatch finder will generate the secret token for you and display it only once after creation, so make sure to write it down in a safe place or copy-paste it into your password management tool.
```
Comment (itamargiv) [inline docs/UserGuide.md:18]: I think this section could benefit from a step by step (numbered list) breakdown on how to (currently) generate a token. We can update this as we progress along with the GUI development.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+## Accessing the API <a id="apiAccess"></a>
+
+To access the API, include your personalised API token as `Bearer <token>` string in the `Authorization:` header field.
```
Comment (itamargiv) [inline docs/UserGuide.md:22]: This repeats what was said before, but I do believe this deserves its own section. Therefore, I think this section belongs before the former section, and its content should be the first two sentences of the former section, that is:

> Once you have logged in and have started to use the Mismatch Finder web interface, you may want to get access to the REST API, as well. You will need a personalised token for that, which you must provide in the `Authorization:` header of your API requests.

**Review by itamargiv — CHANGES_REQUESTED** [review]
:sunglasses: Looks fantastic, thank you for adding this. I've made a couple of editing comments, In addition to the comment I made in [#12](https://github.com/wmde/wikidata-mismatch-finder/pull/12/files#r667892253). Other than that, we are good to go!

File: docs/UserGuide.md [hunk 5]
```diff
@@ -4,19 +4,36 @@
...
+
+**Note:** Don't forget to replace `<your-access-token>` in the example above with your actual personal access token.
+
+## Obtaining an API access token <a id="apiToken"></a>
+
+To obtain a personal access token, follow these steps:
+
+1. At the [application homepage](https://mismatch-finder.toolforge.org/), click the API token link, or go to: https://mismatch-finder.toolforge.org/auth/token
+2. If you do no have an access token already, you will be prompted to create a new token.
+3. **Important!** Write the created token down in a safe place, as it will not be displayed again.
```
Comment (guergana) [inline docs/UserGuide.md:34]: Ah yes. This is good to point out!!!

**Review by guergana — APPROVED** [review]
Looks great for a start. :)

**Pull request merged**